### PR TITLE
Add Continuous Integration Workflow for Github Actions

### PR DESCRIPTION
**Pull request by michhyun1**
**Issue #:**

**Description of changes:**
Introduce Github Actions that will run CI Workflow (build+ integ tests), and execute the deploy script on merges to master.

**Testing**

1. Have you successfully run `npm run build:release` locally? yes
2. How did you test these changes? Cloned this repo into a private repo and tested it there.
3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it? n/a
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? n/a
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? n/a

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    - name: Build
+      run: npm run build:release
+
+  setup_integ_tests:
+    name: Check if Integ Tests are Required
+    needs: build
+    if: github.event_name == 'pull_request_target'
+    runs-on: ubuntu-latest
+    outputs:
+      integration_test_required: ${{ steps.integration_test_required.outputs.needs_integration_test }}
```
Comment (ltrung) [inline .github/workflows/continuous-integration.yml:40]: Does this refer only to the Set integration test requirement to true ?

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        with:
+          fetch-depth: 0
+      - name: Using Node.js 12.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12.x
+      - name: Check if Integ Tests need to Run
+        run: node script/needs-integration-test.js
+        continue-on-error: true
+      - name: Set integration test requirement to true
```
Comment (ltrung) [inline .github/workflows/continuous-integration.yml:53]: Can we append "if necessary" or something similar at the end to make it clear this is not always executed?

File: script/needs-integration-test.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        }
+        if (matches) {
+            requires_integration_test = true;
+        }
+    }
+}
+requires_integration_test ? console.log('requires') : console.log('doesnt require');
+
+requires_integration_test ? process.exit(0) : process.exit(1);
+
```
Comment (ltrung) [inline script/needs-integration-test.js:33]: nit: Extra space.

File: script/needs-integration-test.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+const fs = require('fs');
```
Comment (ltrung) [inline script/needs-integration-test.js:1]: Why do we need this? Cant we just use the existing ruby file or does GitHub VM does not have ruby installed by default?

Comment (michhyun1) [reply script/needs-integration-test.js:1]: Yeah we dont need this. I'll delete it.

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (michhyun1) [reply .github/workflows/continuous-integration.yml]: done

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (michhyun1) [reply .github/workflows/continuous-integration.yml:41]: WE dont need to explicitly set the output to false. We can just set it to true when it is required, and do nothing if its false. The if statement below will evaluate false if this output is not set to true.

**Review by ltrung — APPROVED** [review]

File: .github/workflows/continuous-integration.yml [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+name: CI Workflow
+
+on:
+  pull_request_target:
+    branches:
+      - master
+env:
+  SELENIUM_GRID_PROVIDER: saucelabs
+  CLOUD_WATCH_METRIC: false
```
Comment (devalevenkatesh) [inline .github/workflows/continuous-integration.yml:12]: I guess this will be turned on once testing succeeds here?

File: package.json [hunk 6]
```diff
@@ -32,14 +32,14 @@
...
     "test:retry": "(npm run test:fast || npm run test:fast || npm run test:fast || npm run test:fast || npm run test:fast)",
-    "test:integration": "cd ./integration/js && npm run test && npm run upload-logs-to-s3",
-    "test:integration-audio": "cd ./integration/js && npm run test -- audio && npm run upload-logs-to-s3",
-    "test:integration-video": "cd ./integration/js && npm run test -- video && npm run upload-logs-to-s3",
-    "test:integration-meeting-end": "cd ./integration/js && npm run test -- meeting_end && npm run upload-logs-to-s3",
-    "test:integration-screen-share": "cd ./integration/js && npm run test -- screen_share && npm run upload-logs-to-s3",
-    "test:integration-content-share": "cd ./integration/js && npm run test -- content_share && npm run upload-logs-to-s3",
-    "test:integration-data-message": "cd ./integration/js && npm run test -- data_message && npm run upload-logs-to-s3",
-    "test:integration-meeting-readiness-checker": "cd ./integration/js && npm run test -- meeting_readiness_checker && npm run upload-logs-to-s3"
+    "test:integration": "cd ./integration/js && npm run test ",
```
Comment (devalevenkatesh) [inline package.json:35]: nit: remove trailing spaces.

File: .github/workflows/continuous-integration.yml (hunk 5 above)
Comment (ltrung) [reply .github/workflows/continuous-integration.yml:12]: No need. This is mainly for canary for monitoring.

**Pull request closed without merging**